src/cls/rgw/cls_rgw.cc

```diff
@@ -384,14 +384,18 @@ static void split_key(const string& key, list<string>& vals)
   }
 }
 
-static string escape_str(const string& s)
+static std::string escape_str(const std::string& s)
 {
   int len = escape_json_attr_len(s.c_str(), s.size());
   std::string escaped(len, 0);
   escape_json_attr(s.c_str(), s.size(), escaped.data());
   return escaped;
 }
 
+static std::string quote_escape(const std::string& s) {
+  return '"' + escape_str(s) + '"';
```

Comment:
perhaps its being fussy, but perhaps pre-allocate or reserve escape_str?

Reply:
Not sure if the change is what you had in mind.